torch/distributed/fsdp/flat_param.py

```diff
@@ -442,6 +442,9 @@ def __init__(
         self._training_state = HandleTrainingState.IDLE
         self._debug_level = dist.get_debug_level()
         self._fully_sharded_module = fully_sharded_module
+        # NB: this is updated in the first iteration because optimizers can be applied after
```

Comment:
nit: (in case lazy init happens due to something like loading state dict and not running first forward)
```suggestion
        # NB: this is updated during lazy initialization because optimizers can be applied after
```